src/go/rpk/pkg/cli/cmd/acl/create.go

```diff
@@ -11,255 +11,80 @@ package acl
 
 import (
 	"context"
-	"fmt"
-	"strings"
 
-	"github.com/Shopify/sarama"
-	"github.com/pkg/errors"
-	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
-	"golang.org/x/sync/errgroup"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
 )
 
-func NewCreateACLsCommand(
-	admin func() (sarama.ClusterAdmin, error),
-) *cobra.Command {
-	var (
-		resource        string
-		resourceName    string
-		namePattern     string
-		operations      []string
-		allowPrincipals []string
-		allowHosts      []string
-		denyPrincipals  []string
-		denyHosts       []string
-	)
-	command := &cobra.Command{
-		Use:          "create",
-		Short:        "Create ACLs",
-		SilenceUsage: true,
-		Args: func(_ *cobra.Command, _ []string) error {
-			if len(allowPrincipals) == 0 &&
-				len(denyPrincipals) == 0 {
-				return fmt.Errorf(
-					"at least one of --%s or --%s must be set",
-					allowPrincipalFlag,
-					denyPrincipalFlag,
-				)
+func NewCreateCommand(fs afero.Fs) *cobra.Command {
+	var a acls
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create ACLs.",
+
+		Args: cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := kafka.NewFranzClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer cl.Close()
+
+			creations, err := a.createCreations()
+			out.MaybeDieErr(err)
+			if len(creations) == 0 {
+				out.Exit("Specified flags result in no ACLs to be created.")
 			}
 
-			if resource == "cluster" {
-				resourceName = defaultClusterResourceName
+			req := kmsg.NewPtrCreateACLsRequest()
+			req.Creations = creations
+			resp, err := req.RequestWith(context.Background(), cl)
+			out.MaybeDie(err, "unable to issue acl creation request: %v", err)
+			if len(resp.Results) != len(req.Creations) {
+				out.Die("response contained %d results of the expected %d!", len(resp.Results), len(req.Creations))
```

Comment:
Could we print the difference between result and response in debug mode (`--verbose` global flag)?

Applies to other commands (delete, list).

Reply:
What should this look like? spew.Dump? A json dump? I'm leaning inline JSON.

Reply:
I'll inline-json here, and remove this from delete. It doesn't exist in list.

Reply:
Actually if it's alright, I'd like to punt on adding an inline dump right here, and instead wait until a broader json-output format flag is supported (so this will fail, then it can be re-ran with `-j` or something).

This code path really should not be hit, and it implies a bug in redpanda if it is. It's a sanity check for now, and the inline json  / `--verbose` handling gets very ugly fast.

Reply:
Agree